extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ServerResourceMethodCustomizerBuildItem.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.resteasy.reactive.server.deployment;
+
+import org.jboss.resteasy.reactive.common.model.ResourceClass;
+import org.jboss.resteasy.reactive.server.model.ServerResourceMethod;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * A build item that can be used to customize the built {@link ServerResourceMethod} instances before the
+ * final {@link org.jboss.resteasy.reactive.server.core.DeploymentInfo} is created
+ */
+public final class ServerResourceMethodCustomizerBuildItem extends MultiBuildItem {
+
+    private final ServerResourceMethodCustomizer customizer;
+
+    public ServerResourceMethodCustomizerBuildItem(ServerResourceMethodCustomizer customizer) {
+        this.customizer = customizer;
+    }
+
+    public ServerResourceMethodCustomizer getCustomizer() {
+        return customizer;
+    }
+
+    /**
+     * Implementations will be invoked for each {@link ServerResourceMethod} discovered by Quarkus with the purpose of
+     * altering the ServerResourceMethod
+     */
+    public interface ServerResourceMethodCustomizer {
```

Comment:
IMHO this sort of thing should be in the `indepdent-projects/resteasy-reactive` project, rather than the Quarkus extension.

Reply:
@geoand I had a look at an alternate approach at https://github.com/quarkusio/quarkus/commit/7c1f1f0b505fe41285558b747b90e2eebce9b56b that just expands the existing MethodScanner stuff, WDYT?

Reply:
I was actually trying to force myself to do that but got to lazy 😸.

The reason I initially went down the road I did, was because I thought I needed the deployment path.
I only figured out that I didn't when I finished the implementation ☹️.
Feel free to push a commit including your change, as there is no good reason to go with my approach, all things considered